services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/RecipientProcessor.java

```diff
@@ -38,8 +41,21 @@
      * @return computed recipient that were affected to card
      */
     public ComputedRecipient processAll(CardPublicationData card) {
-        Recipient recipient = card.getRecipient();
-        ComputedRecipient computedRecipient = processAll(recipient);
+        ComputedRecipient computedRecipient;
+        if ((card.getUserRecipients() != null && !card.getUserRecipients().isEmpty())
+                        || (card.getGroupRecipients() != null && !card.getGroupRecipients().isEmpty())) {
+            ComputedRecipient.BuilderEncapsulator result = ComputedRecipient.encapsulatedBuilder();
+            ComputedRecipient.ComputedRecipientBuilder builder = result.builder();
+
+            builder.users(card.getUserRecipients() == null ? Collections.emptyList() : card.getUserRecipients());
+            builder.groups(card.getGroupRecipients() == null ? Collections.emptyList() : card.getGroupRecipients());
+            computedRecipient = builder.build();
+        } else {
+            log.warn("Using deprecated recipient field");
```

Comment:
Change the warning text to : 

_"Use of recipient field is deprecated. Please use usersRecipients and groupsRecipients fields (See documentation)"_